vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/MessageItemFactory.kt

```diff
@@ -197,13 +202,25 @@ class MessageItemFactory @Inject constructor(
             is MessagePollContent -> buildPollItem(messageContent, informationData, highlight, callback, attributes)
             is MessageLocationContent -> buildLocationItem(messageContent, informationData, highlight, attributes)
             is MessageBeaconInfoContent -> liveLocationShareMessageItemFactory.create(params.event, highlight, attributes)
+            is MessageVoiceBroadcastInfoContent -> buildVoiceBroadcastItem(messageContent, highlight, callback, attributes)
             else -> buildNotHandledMessageItem(messageContent, informationData, highlight, callback, attributes)
         }
         return messageItem?.apply {
             layout(informationData.messageLayout.layoutRes)
         }
     }
 
+    private fun getLastMessageContent(event: TimelineEvent): MessageContent? {
+        return with(event) {
+            // Iterate on event types which are not part of the matrix sdk, otherwise fallback to the sdk method
```

Comment:
I think we could extract this function, to make is reusable. Since there are lots of occurence of `event.getLastMessageContent()` in the codebase, we may want to replace all of them with something like:
```kotlin
fun TimelineEvent.getVectorLastMessageContent(): MessageContent? {
... your code
}
```

Reply:
Done https://github.com/vector-im/element-android/pull/7273/commits/b604e1fb712f6adc9c721c214d5e7c3b2c958720 I replaced all the `event.getLasMessageContent` references in the vector module